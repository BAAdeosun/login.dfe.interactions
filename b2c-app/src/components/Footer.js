import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
