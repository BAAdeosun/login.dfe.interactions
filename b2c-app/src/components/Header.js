import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;