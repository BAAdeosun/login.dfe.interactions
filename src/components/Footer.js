import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer