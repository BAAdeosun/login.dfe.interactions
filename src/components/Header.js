import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header