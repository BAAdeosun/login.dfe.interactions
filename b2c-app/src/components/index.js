import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}