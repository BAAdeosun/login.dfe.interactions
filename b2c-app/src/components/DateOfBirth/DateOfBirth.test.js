import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

it('renders without crashing', () => {
    shallow(<components.DateOfBirth />);
});

it('renders correctly without props passed in', () => {
    const tree = renderer
        .create(<components.DateOfBirth />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with errors', () => {
    const tree = renderer.create(<components.DateOfBirth showErrors={true} />);
    tree.root.instance.setState({
        errors: {
            dob: {
                visible: {
                    text: 'this is a test'
                }
            }
        }
    });
    expect(tree.toJSON()).toMatchSnapshot()
});

it('calls validation, sets errors and calls onChange callback', () => {

    const mockOnChangeCallback = jest.fn();

    const wrapper = mount(<components.DateOfBirth showErrors={true} onChange={mockOnChangeCallback} />);
    const dayInput = wrapper.find('#dobDay');
    const monthInput = wrapper.find('#dobMonth');
    const yearInput = wrapper.find('#dobYear');

    const validationSpy = jest.spyOn(wrapper.instance(), 'isValidDob');

    const testDay = '10';
    const testMonth = '11';
    const testYear = '2000';

    let dayChangeEvent = {
        target:
        {
            name: 'dobDay',
            value: testDay
        }
    };
    let monthChangeEvent = {
        target:
        {
            name: 'dobMonth',
            value: testMonth
        }
    };
    let yearChangeEvent = {
        target:
        {
            name: 'dobYear',
            value: testYear
        }
    };

    //default values set
    expect(wrapper.state().dobDay).toBe(null);
    expect(wrapper.state().dobMonth).toBe(null);
    expect(wrapper.state().dobYear).toBe(null);
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter date of birth');

    //simulate entering day
    dayInput.simulate('change', dayChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().dobDay).toEqual(testDay);
    //check error message is set as date is still not complete
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter a valid date of birth');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: null });

    //simulate entering month
    monthInput.simulate('change', monthChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().dobMonth).toEqual(testMonth);
    //check error message is set as date is still not complete
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter a valid date of birth');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: null });

    //simulate entering year
    yearInput.simulate('change', yearChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().dobYear).toEqual(testYear);
    //check error message is cleared as date is valid
    expect(wrapper.state().errors.dob.current.text).toEqual('');
    //onChange callback function passed in has been called with date object
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: new Date(testYear, testMonth - 1, testDay) });

    //simulate entering year that means age is below 13
    yearChangeEvent.target.value = '2019';
    yearInput.simulate('change', yearChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().dobYear).toEqual('2019');
    //check error message is set as date is still not complete
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter a valid date of birth');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: null });

    //simulate entering year that means age is above 65
    yearChangeEvent.target.value = '1950';
    yearInput.simulate('change', yearChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().dobYear).toEqual('1950');
    //check error message is set as date is still not complete
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter a valid date of birth');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: null });

    //simulate deleting part of the year
    yearChangeEvent.target.value = '200';
    yearInput.simulate('change', yearChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().dobYear).toEqual('200');
    //check error message is set as date is still not complete
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter a valid date of birth');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: null });

    //simulate clearing all values
    dayChangeEvent.target.value = '';
    dayInput.simulate('change', dayChangeEvent);
    monthChangeEvent.target.value = '';
    monthInput.simulate('change', monthChangeEvent);
    yearChangeEvent.target.value = '';
    yearInput.simulate('change', yearChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state values have been updated
    expect(wrapper.state().dobDay).toEqual('');
    expect(wrapper.state().dobMonth).toEqual('');
    expect(wrapper.state().dobYear).toEqual('');
    //check error message is set as date is still not complete
    expect(wrapper.state().errors.dob.current.text).toEqual('Enter date of birth');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ dob: null });

});