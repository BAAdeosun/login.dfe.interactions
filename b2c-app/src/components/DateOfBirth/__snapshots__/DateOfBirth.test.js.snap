// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with errors 1`] = `
<div
  className="govuk-form-group govuk-form-group--error"
>
  <fieldset
    aria-describedby="date-of-birth-hint"
    className="govuk-fieldset"
    id="dobFieldset"
    role="group"
  >
    <legend
      className="govuk-fieldset__legend govuk-fieldset__legend--l"
    >
      <label
        className="govuk-label"
      >
        Date of birth
      </label>
    </legend>
    <span
      className="govuk-error-message"
      id="dobError"
    >
      <span
        className="govuk-visually-hidden"
      >
        Error:
      </span>
      this is a test
    </span>
    <span
      className="govuk-hint"
      id="date-of-birth-hint"
    >
      For example, 31 3 1980
    </span>
    <div
      className="govuk-date-input"
      id="date-of-birth"
    >
      <div
        className="govuk-date-input__item"
      >
        <div
          className="govuk-form-group"
        >
          <label
            className="govuk-label govuk-date-input__label"
            htmlFor="dobDay"
          >
            Day
          </label>
          <input
            className="govuk-input govuk-date-input__input govuk-input--width-2"
            id="dobDay"
            inputMode="numeric"
            name="dobDay"
            noValidate={true}
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
        </div>
      </div>
      <div
        className="govuk-date-input__item"
      >
        <div
          className="govuk-form-group"
        >
          <label
            className="govuk-label govuk-date-input__label"
            htmlFor="dobMonth"
          >
            Month
          </label>
          <input
            className="govuk-input govuk-date-input__input govuk-input--width-2"
            id="dobMonth"
            inputMode="numeric"
            name="dobMonth"
            noValidate={true}
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
        </div>
      </div>
      <div
        className="govuk-date-input__item"
      >
        <div
          className="govuk-form-group"
        >
          <label
            className="govuk-label govuk-date-input__label"
            htmlFor="dobYear"
          >
            Year
          </label>
          <input
            className="govuk-input govuk-date-input__input govuk-input--width-4"
            id="dobYear"
            inputMode="numeric"
            name="dobYear"
            noValidate={true}
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`renders correctly without props passed in 1`] = `
<div
  className="govuk-form-group "
>
  <fieldset
    aria-describedby="date-of-birth-hint"
    className="govuk-fieldset"
    id="dobFieldset"
    role="group"
  >
    <legend
      className="govuk-fieldset__legend govuk-fieldset__legend--l"
    >
      <label
        className="govuk-label"
      >
        Date of birth
      </label>
    </legend>
    <span
      className="govuk-hint"
      id="date-of-birth-hint"
    >
      For example, 31 3 1980
    </span>
    <div
      className="govuk-date-input"
      id="date-of-birth"
    >
      <div
        className="govuk-date-input__item"
      >
        <div
          className="govuk-form-group"
        >
          <label
            className="govuk-label govuk-date-input__label"
            htmlFor="dobDay"
          >
            Day
          </label>
          <input
            className="govuk-input govuk-date-input__input govuk-input--width-2"
            id="dobDay"
            inputMode="numeric"
            name="dobDay"
            noValidate={true}
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
        </div>
      </div>
      <div
        className="govuk-date-input__item"
      >
        <div
          className="govuk-form-group"
        >
          <label
            className="govuk-label govuk-date-input__label"
            htmlFor="dobMonth"
          >
            Month
          </label>
          <input
            className="govuk-input govuk-date-input__input govuk-input--width-2"
            id="dobMonth"
            inputMode="numeric"
            name="dobMonth"
            noValidate={true}
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
        </div>
      </div>
      <div
        className="govuk-date-input__item"
      >
        <div
          className="govuk-form-group"
        >
          <label
            className="govuk-label govuk-date-input__label"
            htmlFor="dobYear"
          >
            Year
          </label>
          <input
            className="govuk-input govuk-date-input__input govuk-input--width-4"
            id="dobYear"
            inputMode="numeric"
            name="dobYear"
            noValidate={true}
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;
